### PR TITLE
BUG: index field textfield cursor bounces back to the initial position while typing

### PR DESCRIPTION
**Pull request by Cynthia0710**
Screenshot 2023-01-24 at 16 10 50

### CODE REVIEW

File: lib/controls/doc_field_text_edit.dart [hunk 1]
```diff
@@ -46,6 +46,7 @@ class DocFieldTextEditState extends ConsumerState<DocFieldTextEdit> {
           _formKey.currentState?.validate();
         });
       }
+      ctrl.selection = TextSelection.collapsed(offset: ctrl.text.length);
```
Comment (kovarny) [inline lib/controls/doc_field_text_edit.dart:49]: @wanlok what do you think of this simple solution?
I was thinking we need to find the root cause of the bounce first.

Comment (wanlok) [reply lib/controls/doc_field_text_edit.dart:49]: I tried to comment out the validator logic, but the problem still persist.

Comment (wanlok) [reply lib/controls/doc_field_text_edit.dart:49]: I think the next thing to look at is the ref.watch logic. See if a ref.watch in parent is causing the problem.

Comment (kovarny) [reply lib/controls/doc_field_text_edit.dart:49]: I think we need to add more prints() to understand when the rebuild is happening.

**Pull request closed without merging**